Owncloud iOs Client/InstantUpload/InstantUpload.m

```diff
@@ -0,0 +1,215 @@
+//
+//  InstantUpload.m
+//  Owncloud iOs Client
+//
+//  Created by Jon Schneider on 5/20/16.
+//
+
+#import <CoreLocation/CoreLocation.h>
+#import <Photos/Photos.h>
+
+#import "InstantUpload.h"
+
+#import "AppDelegate.h"
+#import "constants.h"
+
+#import "ManageAppSettingsDB.h"
+#import "ManageUsersDB.h"
+#import "UserDto.h"
+
+#define kPhotoLibraryAccessPermissionDeniedAlert [[UIAlertView alloc] initWithTitle:NSLocalizedString(@"access_photos_library_not_enabled", nil) message:NSLocalizedString(@"message_access_photos_not_enabled", nil) delegate:nil cancelButtonTitle:NSLocalizedString(@"ok", nil) otherButtonTitles:nil]
+
+//TODO: Language should be for background upload, not general Instant Upload
+#define kLocationAccessAlwaysPermissionDeniedAlert [[UIAlertView alloc] initWithTitle:NSLocalizedString(@"location_not_enabled", nil) message:NSLocalizedString(@"message_location_not_enabled", nil) delegate:nil cancelButtonTitle:NSLocalizedString(@"ok", nil) otherButtonTitles:nil]
+
+@interface InstantUpload () <CLLocationManagerDelegate, PHPhotoLibraryChangeObserver>
+
+@property (strong, nonatomic) CLLocationManager *locationManager;
+
+@end
+
+@implementation InstantUpload
+
++ (instancetype) instantUploadManager {
+    static dispatch_once_t once;
+    static InstantUpload *__instantUploadManager;
+    dispatch_once(&once, ^{
+        __instantUploadManager = [[InstantUpload alloc] init];
+    });
+    return __instantUploadManager;
+}
+
+- (instancetype) init {
+    if (self = [super init]) {
+        self.locationManager = [[CLLocationManager alloc] init];
+        self.locationManager.delegate = self;
+        [PHPhotoLibrary.sharedPhotoLibrary registerChangeObserver:self];
+        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(appLifecycleShouldTriggerInstantUpload:) name:UIApplicationWillEnterForegroundNotification object:nil];
+    }
+    return self;
+}
+
+- (void) dealloc {
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (BOOL) enabled {
+    return [ManageUsersDB getActiveUser].instantUpload;
```

Comment:
It is not necessary call to the DataBase everytime that you need something of the Active User. We have it stored on the AppDelegate in memory to prevent many access to the DataBase. the way it is: 
`APP_DELEGATE.activeUser`